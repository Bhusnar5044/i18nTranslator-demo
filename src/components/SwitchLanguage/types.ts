export type Props = {
    // SwitchLanguage prop type
    click: () => void;
};
