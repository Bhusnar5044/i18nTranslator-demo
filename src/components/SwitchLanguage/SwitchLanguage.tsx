import { SingleSelect } from '@components/Lib/SingleSelect/SingleSelect';
import { Options } from '@components/Lib/SingleSelect/types';
// import i18next from 'i18next';
import { ChangeEvent, FC, memo, useState } from 'react';

// const availableLocales = Object.keys(i18next.services.resourceStore.data);

const options: Options = [].map((languageCode) => {
    const nameGenerator = new Intl.DisplayNames(languageCode, { type: 'language' });
    const displayName = nameGenerator.of(languageCode) || '';
    return { value: languageCode, label: displayName };
});

const formattedOptions = [{ value: '', label: 'Switch Language', disabled: true }, ...options];

export const SwitchLanguage: FC = memo(() => {
    const [value, setValue] = useState('');
    const handleChange = (e: ChangeEvent<HTMLSelectElement>) => {
        setValue(e.target.value);
    };
    return <SingleSelect onChange={handleChange} value={value} options={formattedOptions} disableLabel />;
});

SwitchLanguage.displayName = 'SwitchLanguage';
