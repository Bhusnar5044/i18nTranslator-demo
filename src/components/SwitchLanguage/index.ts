export { SwitchLanguage as default } from './SwitchLanguage';
