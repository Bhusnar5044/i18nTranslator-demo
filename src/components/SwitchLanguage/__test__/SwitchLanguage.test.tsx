import { render } from '@test-utils';
import { SwitchLanguage } from '../SwitchLanguage';

describe('SwitchLanguage', () => {
    it('should render properly', () => {
        const { container } = render(<SwitchLanguage />);
        expect(container).toMatchSnapshot();
    });
});
