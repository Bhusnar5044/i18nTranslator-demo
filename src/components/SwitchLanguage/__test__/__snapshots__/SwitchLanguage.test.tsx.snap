// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SwitchLanguage > should render properly 1`] = `
<div>
  <div
    class="form-group"
  >
    <select
      class="sc-eLqIir dmSHxF form-control"
    >
      <option
        class="sc-HzRMQ gxWMSa"
        disabled=""
        value=""
      >
        Switch Language
      </option>
    </select>
  </div>
</div>
`;

exports[`SwitchLanguage should render properly 1`] = `
<div>
  <div
    class="sc-eLqIir dKTRNe"
    gap="1rem"
  >
    <select
      class="sc-dujCx iseyOr"
      title="single-select"
    >
      <option
        class="sc-bFmjqi gTeLlR"
        disabled=""
        value=""
      >
        Switch Language
      </option>
    </select>
  </div>
</div>
`;
